src/Microsoft.ML.TorchSharp/Utils/Index.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+
+namespace System
+{
+    /// <summary>Represent a type can be used to index a collection either from the start or the end.</summary>
+    /// <remarks>
+    /// Index is used by the C# compiler to support the new index syntax
+    /// <code>
+    /// int[] someArray = new int[5] { 1, 2, 3, 4, 5 } ;
+    /// int lastElement = someArray[^1]; // lastElement = 5
+    /// </code>
+    /// </remarks>
+    public readonly struct Index : IEquatable<Index>
```

Comment:
Its internal now. If we don't exclude it will it cause problems? We aren't cross compiling at the moment so we can't exclude it unless we change that. In my testing it didn't seem to cause any problems, but granted I didn't test with range/index anywhere else besides internally.

Reply:
removed the ifdef. Hopefully by the time we would want that back in we can be targeting something newer than .netstandard 2.0 so we could just remove these files.